File: tests/UITests/PageObjectModels/TheInternetPages/AddRemoveElementsPage.js
const { BasePage } = require('./BasePage');

class AddRemoveElementsPage extends BasePage {
  constructor(page) {
    super(page);
    this.url = '/add_remove_elements/';
    this.addElementButton = page.getByRole('button', { name: 'Add Element' });
    this.deleteButtons = page.getByRole('button', { name: 'Delete' });
  }

  async navigate() {
    await this.navigateTo(this.url);
  }

  async clickAddElement() {
    await this.addElementButton.click();
  }

  async clickDeleteButton(index) {
    await this.deleteButtons.nth(index).click();
  }

  async getDeleteButtonCount() {
    return this.deleteButtons.count();
  }
}

module.exports = { AddRemoveElementsPage };

File: tests/UITests/PageObjectModels/TheInternetPages/AddRemoveElementsPage.cs
using Microsoft.Playwright;

namespace Tests.UITests.PageObjectModels.TheInternetPages;

public class AddRemoveElementsPage : BasePage
{
    private const string Url = "https://the-internet.herokuapp.com/add_remove_elements/";

    private ILocator AddElementButton => Page.GetByRole(AriaRole.Button, new() { Name = "Add Element" });
    private ILocator DeleteButtons => Page.GetByRole(AriaRole.Button, new() { Name = "Delete" });

    public AddRemoveElementsPage(IPage page) : base(page) { }

    public async Task Navigate() => await NavigateTo(Url);

    public async Task ClickAddElement() => await AddElementButton.ClickAsync();

    public async Task ClickDeleteButton(int index) => await DeleteButtons.Nth(index).ClickAsync();

    public async Task<int> GetDeleteButtonCount() => await DeleteButtons.CountAsync();
}
